### PR TITLE
[WebApp] Update referral items

### PR DESCRIPTION
**Pull request by Maks19**
**Description:** Update referral items

**Issue:** https://www.notion.so/saladtech/Update-Webapp-update-referral-new-endpoint-d986af559e9844d2a55e80d2fed02318?pvs=23

### CODE REVIEW

File: packages/web-app/src/modules/account-views/referral-views/components/CurrentReferralPanel.tsx [hunk 1]
```diff
@@ -19,22 +17,17 @@ interface Props extends WithStyles<typeof styles> {
...
-            <ReferralCodeEntryContainer />
-          </div>
-        )}
-      </>
-    )
-  }
-}
+const _CurrentReferralPanel = ({ classes, referral }: Props) => (
+  <>
+    {referral && referral.referralDefinition && <CurrentReferralProgress referral={referral} />}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/account-views/referral-views/components/CurrentReferralPanel.tsx:22]:
```suggestion
    {referral?.referralDefinition && <CurrentReferralProgress referral={referral} />}
```

File: packages/web-app/src/modules/account-views/referral-views/components/CurrentReferralProgress.tsx [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-import type { ReactNode } from 'react'
-import { Component } from 'react'
 import type { WithStyles } from 'react-jss'
 import withStyles from 'react-jss'
-import { P, ProgressBar, SectionHeader } from '../../../../components'
 import type { SaladTheme } from '../../../../SaladTheme'
+import { P, ProgressBar, SectionHeader } from '../../../../components'
 import type { Referral } from '../../../referral/models'
-import { percentComplete } from '../../../referral/models'
+import { progressCompletePercentage } from '../../../referral/models'
```
Comment (vitto-moz) [inline packages/web-app/src/modules/account-views/referral-views/components/CurrentReferralProgress.tsx:7]:
```suggestion
import { getProgressCompletePercentage } from '../../../referral/models'
```

File: packages/web-app/src/modules/account-views/referral-views/components/CurrentReferralProgress.tsx [hunk 3]
```diff
@@ -33,33 +31,30 @@ const styles = (theme: SaladTheme) => ({
...
-  public override render(): ReactNode {
-    const { referral, classes } = this.props
-    if (!referral || !referral.referralDefinition) return null
-    return (
-      <div key={referral.refereeId} className={classnames(classes.container)}>
-        <SectionHeader>Your Progress</SectionHeader>
+const _CurrentReferralProgress = ({ classes, referral }: Props) => {
+  const { code, earnedBalance, refereeId, referralDefinition } = referral || {}
+
+  const progressPercentage = progressCompletePercentage(earnedBalance, referralDefinition?.balanceThreshold ?? 1)
```
Comment (vitto-moz) [inline packages/web-app/src/modules/account-views/referral-views/components/CurrentReferralProgress.tsx:40]:
```suggestion
  const progressCompletePercentage= getProgressCompletePercentage(earnedBalance, referralDefinition?.balanceThreshold ?? 1)
```

File: packages/web-app/src/modules/account-views/referral-views/components/ReferralList.tsx [hunk 4]
```diff
@@ -29,7 +29,10 @@ const _ReferralList: FC<Props> = ({ classes, latestReferrals }) => {
       {hasReferrals && (
         <Scrollbars>
           <div className={classes.content}>
-            {latestReferrals && latestReferrals.map((x) => <ReferralItem referral={x} />)}
+            {latestReferrals &&
+              latestReferrals.map(
+                (latestReferral) => latestReferral.referralDefinition && <ReferralItem referral={latestReferral} />,
+              )}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/account-views/referral-views/components/ReferralList.tsx:35]:
```suggestion
            { latestReferrals?.map(
                (latestReferral) => latestReferral.referralDefinition && <ReferralItem referral={latestReferral} />,
              )}
```

Comment (vitto-moz) [reply packages/web-app/src/modules/account-views/referral-views/components/ReferralList.tsx:35]: not really sure React doesn't error it

Comment (Maks19) [reply packages/web-app/src/modules/account-views/referral-views/components/ReferralList.tsx]: It should work
Fixed

**Review by vitto-moz — APPROVED** [review]

**Pull request merged**